app.js

```diff
@@ -380,3 +382,82 @@ const filterByTags = new FilterByTags();
 class Booking1 {}
 
 class Booking2 {}
+
+class FilterByRating {
+  constructor() {
+    this.ratingRadiosRow1 = document.querySelectorAll("[name='rating_row_1']");
+
+    this.ratingRadiosRow2 = document.querySelectorAll("[name='rating_row_2']");
+
+    this.ratingRadiosRow1.forEach((radio) => {
+      radio.addEventListener("change", () => {
+        this.filterChallengesByRating();
+      });
+    });
+
+    this.ratingRadiosRow2.forEach((radio) => {
+      radio.addEventListener("change", () => {
+        this.filterChallengesByRating();
+      });
+    });
+  }
+
+  filterChallengesByRating() {
+    const challengesContainer = document.querySelector("#ourChallenges");
+
+    const api = new APIadapter();
+
+    api.fetchAllChallenges().then((challenges) => {
+      const selectedRatingRow1 = Array.from(this.ratingRadiosRow1).find(
+        (radio) => radio.checked
+      );
+
+      const selectedRatingRow2 = Array.from(this.ratingRadiosRow2).find(
+        (radio) => radio.checked
+      );
+
+      const ratingRow1 = selectedRatingRow1
+        ? parseFloat(selectedRatingRow1.value)
+        : 0;
```

Comment:
Another way of writing this is by using the ["optional chaining"](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) operator and ["nullish coalescing"](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing). Here's what that would look like (note the `?.` instead of `.` and the `??` at the end):

```js
const ratingRow1 = selectedRatingRow1?.value ?? 0;
```

The `?.` (optional chaining) operator means that it will evaluate the expression as `selectedRatingRow1.value` (and try to access `value` on the `selectedRatingRow1` object) only if `selectedRatingRow1` exists, and otherwise it will evaluate the entire expression to `undefined`.

Compare that with using just `selectedRatingRow1.value` which will always try to access `value`, even if `selectedRatingRow1` is `undefined` (doesn't exist).

The `?? 0` at the end means that if the preceeding expression (to the left of `??`) evaluates to undefined, the entire expression will evaluate to zero (the value to the right of `??`).